### PR TITLE
Merge jerry_get_value_without_error and jerry_value_clear_error_flag

### PR DESCRIPTION
**Pull request by imiklos**
JerryScript-DCO-1.0-Signed-off-by: Istvan Miklos imiklos2@inf.u-szeged.hu

### CODE REVIEW

File: docs/02.API-REFERENCE.md [hunk 1]
```diff
@@ -1710,29 +1710,34 @@ jerry_get_error_type (const jerry_value_t value);
...
-## jerry_value_clear_error_flag
+## jerry_get_value_from_error
 
 **Summary**
 
-Clear both the error and abort flags.
+Get the value from an error.
+
+Extract the api value from an error. If the second argument is true
+it will release the the input error value.
```
Comment (rerobika) [inline docs/02.API-REFERENCE.md]: Unnecessary `the`.

File: jerry-core/api/jerry.c [hunk 2]
```diff
@@ -915,18 +915,34 @@ jerry_is_feature_enabled (const jerry_feature_t feature) /**< feature to check *
 } /* jerry_is_feature_enabled */
 
 /**
- * Clear the error flag
+ * Get the value from an error value.
+ *
+ * Extract the api value from an error. If the second argument is true
+ * it will release the the input error value.
```
Comment (rerobika) [inline jerry-core/api/jerry.c]: Ditto.

File: jerry-core/api/jerry.c [hunk 3]
```diff
@@ -915,18 +915,34 @@ jerry_is_feature_enabled (const jerry_feature_t feature) /**< feature to check *
...
+ *
+ * Note:
+ *      returned value must be freed with jerry_release_value, when it is no longer needed.
+ *
+ * @return jerry_value_t value
  */
-void
-jerry_value_clear_error_flag (jerry_value_t *value_p) /**< api value */
+jerry_value_t jerry_get_value_from_error (jerry_value_t value, /**< api value */
+                                          bool release) /**< release value */
```
Comment (zherczeg) [inline jerry-core/api/jerry.c]: comment: release api value

File: tests/unit-core/test-api-errortype.c [hunk 4]
```diff
@@ -40,11 +40,12 @@ main (void)
     TEST_ASSERT (jerry_value_is_error (error_obj));
     TEST_ASSERT (jerry_get_error_type (error_obj) == errors[idx]);
 
-    jerry_value_clear_error_flag (&error_obj);
+    error_obj = jerry_get_value_from_error (error_obj, true);
 
     TEST_ASSERT (jerry_get_error_type (error_obj) == errors[idx]);
 
     jerry_release_value (error_obj);
+
```
Comment (zherczeg) [inline tests/unit-core/test-api-errortype.c]: Remove this newline.

File: tests/unit-core/test-api-set-and-clear-error-flag.c [hunk 5]
```diff
@@ -27,10 +27,12 @@ main (void)
   jerry_value_set_error_flag (&obj_val);
   jerry_value_t err_val = jerry_acquire_value (obj_val);
 
-  jerry_value_clear_error_flag (&obj_val);
-  jerry_release_value (obj_val);
+  obj_val = jerry_get_value_from_error (obj_val, true);
```
Comment (zherczeg) [inline tests/unit-core/test-api-set-and-clear-error-flag.c]: I think the obj_val should be err_val in `jerry_get_value_from_error (obj_val, true);`,

File: tests/unit-core/test-api-set-and-clear-error-flag.c [hunk 6]
```diff
@@ -27,10 +27,12 @@ main (void)
...
 
+  JERRY_ASSERT (obj_val != err_val);
   jerry_release_value (err_val);
+  jerry_release_value (obj_val);
+
```
Comment (zherczeg) [inline tests/unit-core/test-api-set-and-clear-error-flag.c]: Remove this newline.

**Comment by imiklos** [comment]
Thank you for the reviews, I updated the PR with these changes.

File: docs/02.API-REFERENCE.md [hunk 7]
```diff
@@ -1710,29 +1710,34 @@ jerry_get_error_type (const jerry_value_t value);
...
 **Prototype**
 
 ```c
-void
-jerry_value_clear_error_flag (jerry_value_t *value_p);
+jerry_value_t
+jerry_get_value_from_error (jerry_value_t value, bool release)
 ```
 
-- `value_p` - pointer to an api value
```
Comment (LaszloLango) [inline docs/02.API-REFERENCE.md]: - `value` - error (api) value
- `release` - defines whether input api value must be released

File: docs/02.API-REFERENCE.md [hunk 8]
```diff
@@ -1710,29 +1710,34 @@ jerry_get_error_type (const jerry_value_t value);
...
-## jerry_value_clear_error_flag
+## jerry_get_value_from_error
 
 **Summary**
 
-Clear both the error and abort flags.
+Get the value from an error.
+
+Extract the api value from an error. If the second argument is true
+it will release the input error value.
```
[the remark this answers is not in the document]
Comment (LaszloLango) [reply docs/02.API-REFERENCE.md]: My suggestion:
```
Many API functions cannot be called with an error value. This function extracts the API value from an error. The second argument defines whether the input error value must be released or not. If it is set to `true`, then a [`jerry_release_value`](#jerry_release_value) function will be called for the first argument, so the error value won't be available after the call of `jerry_get_value_from_error`. The second argument should be false if both error and its represented value are needed. The first argument is returned unchanged if it is not an error.
```

File: jerry-core/api/jerry.c [hunk 9]
```diff
@@ -915,18 +915,34 @@ jerry_is_feature_enabled (const jerry_feature_t feature) /**< feature to check *
...
+ * it will release the input error value.
+ *
+ * Note:
+ *      returned value must be freed with jerry_release_value, when it is no longer needed.
+ *
+ * @return jerry_value_t value
  */
-void
-jerry_value_clear_error_flag (jerry_value_t *value_p) /**< api value */
+jerry_value_t jerry_get_value_from_error (jerry_value_t value, /**< api value */
```
Comment (LaszloLango) [inline jerry-core/api/jerry.c]: return value must be on separate line:
```
jerry_value_t
jerry_get_value_from_error (jerry_value_t value, /**< api value */
```

File: docs/02.API-REFERENCE.md [hunk 10]
```diff
@@ -1710,29 +1710,34 @@ jerry_get_error_type (const jerry_value_t value);
...
-jerry_value_clear_error_flag (jerry_value_t *value_p);
+jerry_value_t
+jerry_get_value_from_error (jerry_value_t value, bool release)
 ```
 
-- `value_p` - pointer to an api value
-
 **Example**
 
 ```c
```
Comment (LaszloLango) [inline docs/02.API-REFERENCE.md]: Please add an example for a not release error too.

**Comment by LaszloLango** [comment]
Related issue #2213

**Comment by imiklos** [comment]
I updated the PR. Thanks for the reviews.

File: docs/02.API-REFERENCE.md [hunk 11]
```diff
@@ -1710,41 +1710,70 @@ jerry_get_error_type (const jerry_value_t value);
...
+**Example 2**
+
+```c
+{
+  jerry_value_t error;
+  ... // create or acquire value
+
+  jerry_value_set_error_flag (&error);
+  jerry_value_t value = jerry_get_value_from_error (error, false);
+
```
Comment (LaszloLango) [inline docs/02.API-REFERENCE.md]: `... // both 'error' and 'value' can be used and must be released when they are no longer needed`

Comment (imiklos) [reply docs/02.API-REFERENCE.md]: Thanks, updated the PR.

**Pull request merged**